imageio/plugins/tifffile_v3.py

```diff
@@ -302,9 +302,12 @@ def metadata(
                 if flag_value and hasattr(self._fh, flag + "_metadata"):
                     flavor_metadata = getattr(self._fh, flag + "_metadata")
                     if isinstance(flavor_metadata, tuple):
-                        metadata.update(flavor_metadata[0])
-                    else:
+                        flavor_metadata = flavor_metadata[0]
+                    if isinstance(flavor_metadata, dict):
```

Comment:
```suggestion
                    elif isinstance(flavor_metadata, dict):
```

I think this should be elif here, right? because we are deciding how to unpack the metadata depending on it's type.

Reply:
Nope. I slightly adjusted the existing code to unpack a _potential_ dict from the tuple's first entry. Previously, there were two direct *update* calls. Now, if it is a tuple, we get the first entry and replace *flavour_metadata* with it, then actually do check if it is a dict before updating.